.github/workflows/snap-testing.yml

```diff
@@ -55,9 +60,21 @@ jobs:
         uses: actions/checkout@v2
 
       - name: Test snap
+        if: ${{ github.event_name == 'pull_request' || github.event_name == 'workflow_dispatch' }}
         uses: ./test
         with:
           name: ${{matrix.name}}
           channel: ${{matrix.channel}}
           platform_channel: ${{matrix.platform_channel}}
           print_logs: ${{matrix.print_logs}}
+
+      - name: Test snap weekly
+        if: ${{ github.event_name == 'schedule' }}
```

Comment:
Was this conditional step really necessary? We could do full config tests for all events.

If you insist on having full config tests only on schedule, then the `with.full_config_test` itself can take an if statement, e.g. `full_config_test: ${{ github.event.name == `schedule` && true ||  false  }}`. It is called the "fake ternary operator".

Reply:
Yes, I'm okay with having full config tests for all events. Is a longer testing time for each PRs acceptable?

Reply:
Yes, there will be less PRs as we complete the development of tests. Moreover, we can always change the workflow and run what we need when PR is in draft.